src/main/java/org/openstreetmap/atlas/checks/commands/MapRouletteUploadCommand.java

```diff
@@ -0,0 +1,154 @@
+import java.util.Map;
+
+import org.apache.commons.io.FilenameUtils;
+import org.openstreetmap.atlas.checks.maproulette.MapRouletteCommand;
+import org.openstreetmap.atlas.checks.maproulette.MapRouletteConfiguration;
+import org.openstreetmap.atlas.checks.maproulette.data.Challenge;
+import org.openstreetmap.atlas.checks.maproulette.data.Task;
+import org.openstreetmap.atlas.checks.maproulette.serializer.ChallengeDeserializer;
+import org.openstreetmap.atlas.checks.maproulette.serializer.TaskDeserializer;
+import org.openstreetmap.atlas.streaming.resource.File;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+import org.openstreetmap.atlas.utilities.configuration.StandardConfiguration;
+import org.openstreetmap.atlas.utilities.runtime.CommandMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * Given a directory of log files created by atlas-checks, upload those files to MapRoulette.
+ *
+ * @author nachtm
+ */
+public class MapRouletteUploadCommand extends MapRouletteCommand
+{
+    private static final Switch<File> INPUT_DIRECTORY = new Switch<>("logfiles",
+            "Path to folder containing log files to upload to MapRoulette.", File::new,
+            Optionality.REQUIRED);
+    private static final Switch<File> CONFIG_LOCATION = new Switch<>("config",
+            "Path to a file containing MapRoulette challenge configuration.", File::new,
+            Optionality.OPTIONAL, "config/configuration.json");
+    private static final String PARAMETER_CHALLENGE = "challenge";
+    private static final String LOG_EXTENSION = "log";
+    private static final Logger logger = LoggerFactory.getLogger(MapRouletteUploadCommand.class);
+    private final Map<String, Challenge> checkNameChallengeMap;
+
+    public MapRouletteUploadCommand()
+    {
+        super();
+        this.checkNameChallengeMap = new HashMap<>();
+    }
+
+    @Override
+    protected void execute(final CommandMap commandMap,
+            final MapRouletteConfiguration configuration)
+    {
+        final Gson gson = new GsonBuilder().registerTypeAdapter(Task.class,
+                new TaskDeserializer(configuration.getProjectName())).create();
+        final Configuration instructions = this.loadConfiguration(commandMap);
+        ((File) commandMap.get(INPUT_DIRECTORY)).listFilesRecursively().forEach(logFile ->
+        {
+            if (FilenameUtils.getExtension(logFile.getName()).equals(LOG_EXTENSION))
+            {
+                try (BufferedReader reader = new BufferedReader(new FileReader(logFile.getPath())))
+                {
+                    reader.lines().forEach(line ->
+                    {
+                        final Task task = gson.fromJson(line, Task.class);
+                        try
+                        {
+                            this.addTask(this.getChallenge(task.getChallengeName(), instructions),
```

Comment:
this function doesn't upload anything to maproulette correct? And if not, are you worried about there performance of trying to upload too many tasks at once if there are a large numbers of files/lines

Reply:
I think my comment disappeared somehow so apologies if you're getting double-notified.

The actual upload happens a little further down. I could potentially do an upload after each file -- assuming that MR treats a new upload to the same challenge as an append rather than a replace. If MR handles new uploads by replacing, then I think we have to read in the entire folder before uploading, since results from a particular check could be strewn over multiple files. I can look into that soon and get back to you

Reply:
@jklamer What do you think about this? Just tested on a couple of files and I can confirm that the end result is correct.

Uploading after each file wipes local memory clean so loading a large number of files should be manageable, and I think each file is generally small enough in size that we won't need to worry about splitting there